build/config/linux/libc.gni

```diff
@@ -0,0 +1,29 @@
+# Copyright (c) 2023 The Flutter Authors. All rights reserved.
+# Use of this source code is governed by a BSD-style license that can be
+# found in the LICENSE file.
+
+declare_args() {
+  # The C library used by the compilation host.
+  # Expecting one of: "gnu" (glibc), "uclibc", "musl"
+  host_libc = ""
+
+  # The C library of the compilation target.
+  # Expecting one of: "gnu" (glibc), "uclibc", "musl"
+  target_libc = ""
+}
+
+if (host_libc == "") {
+  # XXX: actually detecting it is complicated
+  host_libc = "gnu"
+}
+if (target_libc == "") {
+  target_libc = host_libc
+}
+
+is_cross_libc = host_libc != target_libc
+
+is_glibc = target_libc == "gnu"
+is_musl = target_libc == "musl"
```

Comment:
Here and below - should we be adding `is_uclibc`?

Reply:
I'm not sure whether there actually are any users. I've listed it in the comment because I've seen some references but they're probably copied from elsewhere